Add meta registrations and distributed decomps: _foreach_div_.Scalar, sqrt_.default

janeyx99 opened the pull request:
This PR unblocks #104780 by resolving spmd tracing test issues and by adding meta registrations for foreach inplace ops (div_ and sqrt_)

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104780
* __->__ #104779
* #104669

janeyx99 committed:
Add meta registrations and distributed decomp _foreach_div_.Scalar

[ghstack-poisoned]

awgu commented on torch/distributed/_spmd/api.py:
```diff
@@ -271,6 +271,9 @@ def _fused_adam_decomp(
     aten._foreach_mul_.Scalar: partial(
         _foreach_binop_scalar_decomp, aten._foreach_mul.Scalar
     ),
```
Are we actively developing SPMD? 😮

janeyx99 replied:
Unsure but I had a distributed test failure when I updated foreach Adam to use inplace div and this should fix.

If this is no longer needed, we should remove the tests 😛 and the code 😛 which I'm thinking would be a bigger more involved change than this.

fegin replied:
Let me disable the tests.

fegin reviewed (approved):
LGTM, I'll disable _spmd tests.

janeyx99 committed:
Update on "Add meta registrations and distributed decomps: _foreach_div_.Scalar, sqrt_.default"

This PR unblocks #104780 by resolving spmd tracing test issues and by adding meta registrations for foreach inplace ops (div_ and sqrt_)




[ghstack-poisoned]

janeyx99 committed:
Update on "Add meta registrations and distributed decomps: _foreach_div_.Scalar, sqrt_.default"

This PR unblocks #104780 by resolving spmd tracing test issues and by adding meta registrations for foreach inplace ops (div_ and sqrt_)




[ghstack-poisoned]
